corporate/lib/stripe.py

```diff
@@ -595,33 +595,72 @@ def update_or_create_stripe_customer(self, payment_method: Optional[str] = None)
 
     def create_stripe_payment_intent(
         self, price_per_license: int, licenses: int, metadata: Dict[str, Any]
-    ) -> PaymentIntent:
+    ) -> str:
+        # NOTE: This charges users immediately.
         customer = self.get_customer()
         assert customer is not None and customer.stripe_customer_id is not None
         payment_intent_data = self.get_data_for_stripe_payment_intent(price_per_license, licenses)
-        stripe_payment_intent = stripe.PaymentIntent.create(
-            amount=payment_intent_data.amount,
-            currency="usd",
+        # Ensure customers have a default payment method set.
+        stripe_customer = stripe_get_customer(customer.stripe_customer_id)
+        if not stripe_customer_has_credit_card_as_default_payment_method(stripe_customer):
+            raise BillingError(
+                "no payment method",
+                _("Please add a credit card before upgrading."),
```

Comment:
I think we're not tagging `BillingError` for translation.

Reply:
Yeah, no reason to translate this one bit.